pandas/io/parsers.py

```diff
@@ -786,17 +786,25 @@ def read(self, nrows=None):
             return ret
 
         # May alter columns / col_dict
-        index, columns, col_dict = self._create_index(ret)
+        index, columns, mappings, col_dict = self._create_index(ret)
 
-        df = DataFrame(col_dict, columns=columns, index=index)
+        if not mappings:
```

Comment:
Explain to me how your list of lists would correctly and efficiently handle this case:
~~~python
>>> data = """1,2,3,4,5
6,7,8,9,10"""
>>> read_csv(StringIO(data), names=['foo', 'bar', 'baz', 'foo', 'lol'],
             parse_dates={'foo_date':[0, 3], 'lol_bar_lol': ['bar', 'lol]})
~~~